zebra-chain/src/sprout/tree.rs

```diff
@@ -13,8 +13,60 @@
 
 use std::fmt;
 
+use byteorder::{ByteOrder, LittleEndian};
+use lazy_static::lazy_static;
 #[cfg(any(test, feature = "proptest-impl"))]
 use proptest_derive::Arbitrary;
+use sha2::digest::generic_array::GenericArray;
+
+use super::commitment::NoteCommitment;
+
+const MERKLE_DEPTH: usize = 29;
+
+/// MerkleCRH^Sprout Hash Function
+///
+/// MerkleCRH^Sprout(layer, left, right) := SHA256Compress(left || right)
+///
+/// `layer` is unused for Sprout but used for the Sapling equivalent.
+///
+/// https://zips.z.cash/protocol/protocol.pdf#merklecrh
+fn merkle_crh_sprout(left: [u8; 32], right: [u8; 32]) -> [u8; 32] {
+    let mut other_block = [0u8; 64];
+    other_block[..32].copy_from_slice(&left[..]);
+    other_block[32..].copy_from_slice(&right[..]);
+
+    let mut state = [0u32; 8];
```

Comment:
This needs to be the SHA-256 initialization vector: https://github.com/RustCrypto/hashes/blob/0b9b8f5aee168f8eb9ac59a8932c3adf947cb406/sha2/src/consts.rs#L170-L172

Reply:
OH, DAMN

Reply:
Done 👯